hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java

```diff
@@ -127,17 +129,17 @@ private void persistAccount(Address address, long nonce, Wei balance) {
                         ? Id.fromGrpcAccount(accountIdFromEvmAddress(resolvedAddress))
                         : Id.fromGrpcAccount(accountIdFromEvmAddress(address)),
                 0L,
-                balance.toLong(),
+                Suppliers.memoize(() -> balance.toLong()),
                 false,
-                0L,
+                () -> 0L,
                 0L,
                 null,
                 0,
-                Collections.emptySortedMap(),
-                Collections.emptySortedMap(),
-                Collections.emptySortedSet(),
-                0,
-                0,
+                TreeMap::new,
+                TreeMap::new,
+                TreeSet::new,
```

Comment:
Prefer zero allocation collections unless needed.
```suggestion
                Collections::emptySortedMap,
                Collections::emptySortedMap,
                Collections::emptySortedSet,
```